client/src/main/scala/drt/client/SPAMain.scala

```diff
@@ -23,13 +25,16 @@ object SPAMain {
   sealed trait Loc
 
   case class TerminalPageTabLoc(
-                                 terminal: String,
-                                 mode: String = "current",
-                                 subMode: String = "desksAndQueues",
-                                 date: Option[String] = None,
-                                 timeRangeStartString: Option[String] = None,
-                                 timeRangeEndString: Option[String] = None
-                               ) extends Loc {
+                                  terminal: String,
+                                  mode: String = "current",
+                                  subMode: String = "desksAndQueues",
+                                  queryParams: Map[String, String] = Map.empty[String, String]
+                                ) extends Loc {
+    val date: Option[String] = queryParams.get("date").filter(_.matches(".+"))
+    val timeRangeStartString: Option[String] = queryParams.get("timeRangeStart").filter(_.matches("[0-9]+"))
+    val timeRangeEndString: Option[String] = queryParams.get("timeRangeEnd").filter(_.matches("[0-9]+"))
+    val viewType: ViewType = queryParams.get("viewType").map(vt => if (ViewRecs.toString.contains(vt)) ViewRecs else ViewDeps).getOrElse(ViewDeps)
```

Comment:
It feels a little brittle to do a toString.contains to check against a string. Could we do something like introduce a `queryParamValue` which is then used as both the value to check the query parameter against and the value to add to the url ?

Reply:
@richbirch Sorry, I understand what you mean here.

Reply:
I mean add a function or val to ViewRecs & ViewDeps called queryParamValue which returns the definitive string expected in the URL. It should also be used to add the string to the URL. This guards against the routing breaking should someone rename the ViewDeps or ViewRecs classes in future